File: 20. Valid Parentheses/20. Valid Parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []
        brackets = {')': '(', '}': '{', ']': '['}
        
        for char in s:
            if char in '({[':
                stack.append(char)
            elif char in ')}]':
                if not stack or stack.pop() != brackets[char]:
                    return False
        
        return len(stack) == 0

File: 20. Valid Parentheses/20. Valid Parentheses.cs
public class Solution {
    public bool IsValid(string s) {
        Stack<char> stack = new Stack<char>();
        foreach (char c in s) {
            if (c == '(' || c == '{' || c == '[') {
                stack.Push(c);
            } else {
                if (stack.Count == 0) return false;
                if ((c == ')' && stack.Peek() != '(') ||
                    (c == '}' && stack.Peek() != '{') ||
                    (c == ']' && stack.Peek() != '[')) {
                    return false;
                }
                stack.Pop();
            }
        }
        return stack.Count == 0;
    }
}